chain-signatures/node/src/protocol/sync/mod.rs

```diff
@@ -178,25 +184,59 @@ async fn broadcast_sync(
         .join_all()
         .await
         .into_iter()
-        .filter_map(|(p, view)| if let Ok(()) = view { Some(p) } else { None })
+        .filter_map(|(p, view)| {
+            if let Ok(view) = view {
+                Some((p, view))
+            } else {
+                None
+            }
+        })
         .collect::<Vec<_>>();
 
     tracing::info!(
         elapsed = ?start.elapsed(),
-        responded = ?resps,
+        responded = ?resps.iter().map(|(p, _)| *p).collect::<Vec<_>>(),
         "broadcast completed",
     );
 
+    let mut triple_kick = HashMap::new();
+    for &id in &update.triples {
+        let entry = triple_kick.entry(id).or_insert_with(Vec::new);
+        for (p, view) in &resps {
+            if !view.triples.contains(&id) {
+                entry.push(*p);
+            }
+        }
+    }
+    triples.kick_participants(triple_kick).await;
+
+    let mut presignature_kick = HashMap::new();
+    for &id in &update.presignatures {
+        let entry = presignature_kick.entry(id).or_insert_with(Vec::new);
```

Comment:
nit: could use `or_default`

```suggestion
        let entry = presignature_kick.entry(id).or_default();
```

keeping it as-is is also fine